pkg/generate/dockercompose/generate.go

```diff
@@ -0,0 +1,537 @@
+package dockercompose
+
+import (
+	"fmt"
+	"net"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	utilerrs "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	"github.com/openshift/origin/pkg/generate/git"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	dockerfileutil "github.com/openshift/origin/pkg/util/docker/dockerfile"
+	"github.com/openshift/origin/third_party/github.com/docker/libcompose/project"
+)
+
+func IsPossibleDockerCompose(path string) bool {
+	switch base := filepath.Base(path); {
+	case base == "docker-compose.yaml", base == "docker-compose.yml":
+		return true
+	default:
+		return false
+	}
+}
+
+// Generate accepts a set of Docker compose project paths and converts them in an
+// OpenShift template definition.
+func Generate(paths ...string) (*templateapi.Template, error) {
+	for i := range paths {
+		path, err := filepath.Abs(paths[i])
+		if err != nil {
+			return nil, err
+		}
+		paths[i] = path
+	}
+	var bases []string
+	for _, s := range paths {
+		bases = append(bases, filepath.Dir(s))
+	}
+
+	context := &project.Context{
+		ComposeFiles: paths,
+	}
+	p := project.NewProject(context)
+	if err := p.Parse(); err != nil {
+		return nil, err
+	}
+	template := &templateapi.Template{}
+	template.Name = p.Name
+
+	serviceOrder := sets.NewString()
+	warnings := make(map[string][]string)
+	for k, v := range p.Configs {
+		serviceOrder.Insert(k)
+
+		var msg string
+		switch {
```

Comment:
seems like this shouldn't be a switch so you can get all the warning messages for a particular config, instead of just one.